components/eamxx/src/physics/p3/impl/p3_conservation_impl.hpp

```diff
@@ -7,6 +7,66 @@
 namespace scream {
 namespace p3 {
 
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::cloud_water_conservation(const Spack& qc, const Scalar dt,
+  Spack& qc2qr_autoconv_tend, Spack& qc2qr_accret_tend, Spack &qc2qi_collect_tend, Spack& qc2qi_hetero_freeze_tend, 
+  Spack& qc2qr_ice_shed_tend, Spack& qc2qi_berg_tend, Spack& qi2qv_sublim_tend, Spack& qv2qi_vapdep_tend,
+  Spack& qcheti_cnt, Spack& qicnt, const bool& use_hetfrz_classnuc, const Spack& cld_frac_l, const Spack& cld_frac_i, const Smask& context)
+{
+
+  Spack sinks;
+  if(use_hetfrz_classnuc){
+    sinks = (qc2qr_autoconv_tend+qc2qr_accret_tend+qc2qi_collect_tend+qcheti_cnt+qc2qr_ice_shed_tend+qc2qi_berg_tend)*dt; // Sinks of cloud water
+  }
+  else{
+    sinks = (qc2qr_autoconv_tend+qc2qr_accret_tend+qc2qi_collect_tend+qc2qi_hetero_freeze_tend+qc2qr_ice_shed_tend+qc2qi_berg_tend)*dt; // Sinks of cloud water
+  }
+  const auto sources = qc; // Source of cloud water
+  const auto il_cldm = min(cld_frac_i,cld_frac_l); 
+  Spack ratio;
+
+  constexpr Scalar qtendsmall = C::QTENDSMALL;
+  Smask enforce_conservation  = sinks > sources && sinks >= qtendsmall && context;  // determine if  conservation corrction is necessary
+  Smask nothing_todo = !enforce_conservation && context;
+
+  if (enforce_conservation.any()){
+    ratio.set(enforce_conservation, sources/sinks);
+    qc2qr_autoconv_tend.set(enforce_conservation, qc2qr_autoconv_tend*ratio);
+    qc2qr_accret_tend.set(enforce_conservation, qc2qr_accret_tend*ratio);
+    qc2qi_collect_tend.set(enforce_conservation, qc2qi_collect_tend*ratio);
+    if(use_hetfrz_classnuc){
+         qcheti_cnt.set(enforce_conservation, qcheti_cnt*ratio);
```

Comment:
Some minor code formatting stuff. 4-space indents are being used here instead of the standard 2. Also, many of the `if` and `else` statements are not formatted like the rest of eamxx. They should be
`if<space>(<condition>)<space>{`
and
`else<space>{`

Reply:
Since this function is overloaded, it leads to code duplication. To avoid that, we may need to restructure the logic (either using helper functions with core functionality or using optional arguments). I am not sure how much effort that is or if it should be done in this PR.